projects/control-service/projects/pipelines_control_service/src/test/java/com/vmware/taurus/authorization/KerberosAuthenticationIT.java

```diff
@@ -0,0 +1,106 @@
+/*
+ * Copyright 2021 VMware, Inc.
+ * SPDX-License-Identifier: Apache-2.0
+ */
+package com.vmware.taurus.authorization;
+
+import com.vmware.taurus.ControlplaneApplication;
+import com.vmware.taurus.KerberosUtil;
+import com.vmware.taurus.service.JobsRepository;
+import com.vmware.taurus.service.model.DataJob;
+import com.vmware.taurus.service.model.JobConfig;
+import org.apache.kerby.kerberos.kerb.KrbException;
+import org.junit.jupiter.api.AfterAll;
+import org.junit.jupiter.api.AfterEach;
+import org.junit.jupiter.api.Assertions;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.TestInstance;
+import org.junit.jupiter.api.extension.ExtendWith;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.boot.test.context.SpringBootTest;
+import org.springframework.boot.web.server.LocalServerPort;
+import org.springframework.test.context.TestPropertySource;
+import org.springframework.test.context.junit.jupiter.SpringExtension;
+
+import java.net.HttpURLConnection;
+import java.net.URL;
+
+/**
+ * This test is in the unit tests directory, because:
+ * A) it's fast (it takes a few seconds)
+ * B) It mocks and manages all its dependencies and does not create side effects .
+ * C) It does not work well with MiniKDC e.g
+ * Our integration tests use the MiniKdc server provided
+ * by spring which as of today hasn't received any updates
+ * in a long time and is problematic. However, minikdc leaves
+ * some files and configurations that interfere with apache kerby
+ * and both tests cannot run on the same runner (even after invoking
+ * the provided cleanup mehtods by the libraries). We should look into
+ * migrating to apache kerby for our integration tests too.
+ */
+@TestPropertySource(properties = {
+      "test.dir=target",
+      "featureflag.security.enabled=true",
+      "datajobs.security.kerberos.enabled=true",
+      "featureflag.authorization.enabled=false",
+      "datajobs.security.kerberos.kerberosPrincipal=HTTP/localhost",
+      "datajobs.security.kerberos.keytabFileLocation=target/foo.keytab",
+      "datajobs.security.kerberos.krb5ConfigLocation=target/krb5.conf",
+      "java.security.krb5.conf=target/krb5.conf",
+      "sun.security.krb5.debug=true",
+      "datajobs.authorization.authorized-roles="
+})
+@SpringBootTest(webEnvironment = SpringBootTest.WebEnvironment.RANDOM_PORT, classes = ControlplaneApplication.class)
+@ExtendWith(SpringExtension.class)
+@TestInstance(TestInstance.Lifecycle.PER_CLASS)
+public class KerberosAuthenticationIT extends TestKerberosServer {
```

Comment:
I prefer composition over inheritance. I'd have written it as Junit extension. But that's also fine. We've spent enough time on this.